sdk/servicebus/azure-servicebus/migration_guide.md

```diff
@@ -1,164 +1,444 @@
-| `QueueClient.from_connection_string(idle_timeout=None)` | `ServiceBusClient.from_connection_string.get_queue_receiver(max_wait_time=None)` | [providing a timeout](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/servicebus/azure-servicebus/samples/sync_samples/session_pool_receive.py) |
+In V0.50:
 
-### Sub-client initialization
+```python
+client = ServiceBusClient.from_connection_string(connstr)
 
-In v0.50, to send or receive messages, a `QueueClient` would be created directly or from the `ServiceBusClient.get_queue` method,
-after which user would call `get_sender` or `get_receiver` methods. Alternatively, `azure.servicebus.send_handler.Sender` and `azure.servicebus.receive_handler.Receiver` were used.
+queue_client = client.get_queue(queue_name)
+queue_sender = queue_client.get_sender()
+queue_receiver = queue_client.get_receiver()
+queue_dead_letter_receiver = queue_client.get_deadletter_receiver()
 
-In v7.0, these are migrated to `ServiceBusSender` and `ServiceBusReceiver`. Note that these are internal sub-clients and should be initialized from the root `ServiceBusClient` as follows.
+topic_client = client.get_topic(topic_name)
+topic_sender = topic_client.get_sender()
 
+subscription_client = client.get_subscription(topic_name, subscription_name)
+subscription_receiver = subscription_client.get_receiver()
+subscription_dead_letter_receiver = subscription_client.get_deadletter_receiver()
+```
+
+In V7:
 ```Python
 with ServiceBusClient.from_connection_string(connstr) as client:
-    with client.get_queue_sender(queue_name) as sender:
-        # Sending a single message
-        single_message = ServiceBusMessage("Single message")
-        sender.send_messages(single_message)
+    queue_sender = client.get_queue_sender(queue_name)
+    queue_receiver = client.get_queue_receiver(queue_name)
+    queue_dead_letter_receiver = client.get_queue_receiver(queue_name, sub_queue=ServiceBusSubQueue.DEAD_LETTER)
 
-with ServiceBusClient.from_connection_string(connstr) as client:
-    with client.get_queue_receiver(queue_name, max_wait_time=30) as receiver:
-        messages = receiver.receive_messages()
+    topic_sender = client.get_topic_sender(topic_name)
+
+    subscription_receiver = client.get_subscription_receiver(topic_name, subscription_name)
+    subscription_dead_letter_receiver = client.get_subscription_receiver(topic_name, subscription_name, sub_queue=ServiceBusSubQueue.DEAD_LETTER)
+```
+
+### Sending messages
+
+- `send` method is renamed to `send_messages` on `ServiceBusSender` to be consistent in usage of the `messages` suffix in other methods on the receiver and the sender.
+- `BatchMessage` is replaced with `ServiceBusMessageBatch` to send a batch of messages in one call not exceeding the size limit of the sender.
```

Comment:
The main thing to notice for the user is not that there has been a rename from `send` to `send_messages`. I believe that would be straight forward for them to notice. The main catch is that if they want to send an array of messages, previously they need to construct the `BatchMessage`, but now the same send method that takes a single message will also take the array of messages directly.

> `BatchMessage` is replaced with `ServiceBusMessageBatch` to send a batch of messages 

That is not accurate... `BatchMessage` is simply removed. Previously one needed to create it by passing an array of messages, now they can pass the array of messages directly to the send method. `ServiceBusMessageBatch` is an entirely new concept..

```suggestion
- The `send` method is renamed to `send_messages` following the pattern of using the `messages` suffix in methods that deal with messages.
- You can now pass an array of messages directly in method used to send messages rather than first creating a `BatchMessage` 
- Sending multiple messages in a single go always had the potential to fail if batch size exceeded the size limit. To help with this, we have a new class `ServiceBusMessageBatch` which helps in creating a batch that will never increase the size limit.
```